test_run.lua

```diff
@@ -65,9 +65,10 @@ local function wait_lsn(self, waiter, master)
     local sid = self:get_server_id(master)
     local lsn = self:get_lsn(master, sid)
 
-    while self:get_lsn(waiter, sid) < lsn do
+    repeat
         fiber.sleep(0.001)
-    end
+        local waiter_lsn = self:get_lsn(waiter, sid)
+    until waiter_lsn ~= nil and waiter_lsn >= lsn
```

Comment:
(Regarding the code itself.) This way we sleep first, then make attempt to get the LSN. It is better to check LSN first and sleep if the attempt is unsuccessful:

```lua
local remote_lsn = self:get_lsn(waiter, sid)
while remote_lsn == nil or remote_lsn < lsn do
    fiber.sleep(0.001)
    remote_lsn = self:get_lsn(waiter, sid)
end
```

Reply:
Ok, let's use your variant.

Reply:
> (Regarding the code itself.) This way we sleep first, then make attempt to get the LSN. It is better to check LSN first and sleep if the attempt is unsuccessful:
> 
> ```lua
> local remote_lsn = self:get_lsn(waiter, sid)
> while remote_lsn == nil or remote_lsn < lsn do
>     fiber.sleep(0.001)
>     remote_lsn = self:get_lsn(waiter, sid)
> end
> ```

And now you get lsn, sleep and after do the check. Why do you think it's better? As for me I am comfortable with both.

Reply:
In the first variant, you are more likely to get a larger lsn.

Reply:
> Why do you think it's better?

No sleep when suitable data is already available.

Reply:
> > Why do you think it's better?
> 
> No sleep when suitable data is already available.

Ok. Sorry, I think you are suggesting to simply swap `sleep` with `get_lsn`.